libbpf-tools/trace_helpers.c

```diff
@@ -159,6 +166,108 @@ const struct ksym *ksyms__get_symbol(const struct ksyms *ksyms,
 	return NULL;
 }
 
+struct partitions {
+	struct list_head list;
+};
+
+static int partitions__add_partition(struct partitions *partitions,
+				     const char *name, unsigned int dev)
+{
+	struct partition *partition;
+
+	partition = malloc(sizeof(*partition));
```

Comment:
oh come one, why are you jumping from one extreme to another? Is linked list really necessary? Please stop adding unnecessary "infrastructure" code. Use realloc for dynamically grown array, it's much easier and nicer to work with, compared to linked list.  Do use strdup() for name, my previous objection was against insisting on having a single array of all concatenated strings, that's simply not worth it. But why linked list?... Just an array of struct partitions, you can just `realloc(p, (N+1) * sizeof(struct partition))`. Nice and simple.

Reply:
@anakryiko Oh so sorry, I find that I often complicate things,  I will think more to avoid this. For now I have changed to dynamic array.  PTAL again. Thank you for your patient guidance.